File: algo_2021/w4.-queue-bfs/problems/457.Implement_Queue_by_Linked_List/457_implement_queue_by_linked_list__array.py
class MyQueue:
    """
    @param: item: An integer
    @return: nothing
    """
    def __init__(self):
        self.MAXSIZE = 10000000
        self.queue = [0] * self.MAXSIZE
        self.head, self.tail = 0, 0

    def enqueue(self, item):
        # write your code here
        # if queue full
        if self.tail == self.MAXSIZE:
            return
        self.queue[self.tail] = item
        self.tail += 1
        

    """
    @return: An integer
    """
    def dequeue(self):
        # write your code here
        # if queue empty
        if self.head == self.tail:
            return -1
        item = self.queue[self.head]
        self.head += 1 
        return item

File: algo_2021/w4.-queue-bfs-graph/problems/457.Implement_Queue_by_Linked_List/457_implement_queue_by_linked_list__array.java
public class MyQueue {
    /*
     * @param item: An integer
     * @return: nothing
     */

    int MAXSIZE = 100000;
    int[] queue = new int[MAXSIZE];
    int head = 0, tail = 0;
    
    public void enqueue(int item) {
        // write your code here
        // if queue full
        if (tail == MAXSIZE) {
            return; 
        }
        queue[tail++] = item;
    }

    /*
     * @return: An integer
     */
    public int dequeue() {
        // write your code here
        // if queue empty 
        if (head == tail) {
            return -1;
        }
        int item = queue[head++];
        return item; 
    }
}